e2e/cypress/integration/bot_accounts/managing_bot_accounts_spec.js

```diff
@@ -0,0 +1,302 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @bot_accounts
+
+import {generateRandomUser} from '../../support/api/user';
+
+describe('Managing bot accounts', () => {
+    let newTeam;
+    let botName;
+
+    beforeEach(() => {
+        botName = 'bot-' + Date.now();
+
+        // # Set ServiceSettings to expected values
+        const newSettings = {
+            ServiceSettings: {
+                EnableBotAccountCreation: true,
+                DisableBotsWhenOwnerIsDeactivated: true,
+            },
+            PluginSettings: {
+                Enable: true,
+                RequirePluginSignature: false,
+            },
+        };
+        cy.apiUpdateConfig(newSettings);
+
+        // # Create a test bot
+        cy.apiCreateBot(botName, 'Test Bot', 'test bot');
+
+        // # Create and visit new channel
+        cy.apiInitSetup().then(({team}) => {
+            newTeam = team;
+        });
+        cy.apiAdminLogin();
+    });
+
+    it('MM-T1851 No option to create BOT accounts when Enable Bot Account Creation is set to False.', () => {
+        // # Visit bot config
+        cy.visit('/admin_console/integrations/bot_accounts');
+
+        // # Click 'false' to disable
+        cy.findByTestId('ServiceSettings.EnableBotAccountCreationfalse').click();
+
+        // # Save
+        cy.findByTestId('saveSetting').should('be.enabled').click();
+
+        // # Visit the integrations
+        cy.visit(`/${newTeam.name}/integrations/bots`);
+
+        // * Assert that adding bots is not possible
+        cy.get('#addBotAccount').should('not.be.visible');
+    });
+
+    it('MM-T1852 Bot creation via API is not permitted when Enable Bot Account Creation is set to False', () => {
+        // # Visit bot config
+        cy.visit('/admin_console/integrations/bot_accounts');
+
+        // # Click 'false' to disable
+        cy.findByTestId('ServiceSettings.EnableBotAccountCreationfalse').click();
+
+        // # Save
+        cy.findByTestId('saveSetting').should('be.enabled').click();
+
+        // * Validate that creating bot fails
+
+        cy.request({
+            headers: {'X-Requested-With': 'XMLHttpRequest'},
+            url: '/api/v4/bots',
+            method: 'POST',
+            failOnStatusCode: false,
+            body: {
+                username: 'bot-' + Date.now(),
+                display_name: 'test bot',
+                description: 'test bot',
+            },
+        }).then((response) => {
+            expect(response.status).to.equal(403);
```

Comment:
Please add to satisfy the expectation in test case:
``expect(response.body.message).to.equal('Bot creation has been disabled.');``